test/doctest.h

```diff
@@ -0,0 +1,6816 @@
+// ====================================================================== lgtm [cpp/missing-header-guard]
```

Comment:
For what it's worth, do we need to keep this thing up to date? If so how much worse will this look like if we submodule this thing from it's repo?

Reply:
I don't think we need to keep it up to date tbh.